### PR TITLE
262172: Configure KMS for Cluster with etcd Encryption Using KMS V2

### PR DESCRIPTION
**Pull request by aadev1**
The change will enable KMS V2 on the AKS Cluster.  Enabling KMS V2 will give enhanced data security and compliance through KMS-based etcd encryption.

This PR includes:

- Creation of a Platform KeyVault per environment, which will initially be used for storing the KMS Key.

- Pipeline parameter to initialize or rotate the KMS Key

- When the 'initializeOrRotateKmsKey' parameter is selected:
1. A new Key will be created in the Platform KeyVault
2. KeyVaultCryptoUser and KeyVaultContributor roles are assigned to the Cluster managed identity to allow it to use the Key in the private KeyVault
3. customerManagedKey parameters will be provided to the managed-cluster bicep module

- A new postdeploy script (Set-AksSecrets.ps1) has been created to update all secrets on the cluster, so they are encrypted with the KMS Key.
  The script will temporarily add 'Azure Kubernetes Service RBAC Writer' permission to the ADO SPN to allow it to update the secrets.  The permissions will be removed at the end of the script.  In order to grant the permissions the lock also has to be removed from the AKS cluster temporarily and then added back on.

[AB#262172](https://dev.azure.com/defragovuk/93c65135-d16b-49d6-a191-4a5313532779/_workitems/edit/262172)

# Testing
https://dev.azure.com/defragovuk/DEFRA-FFC/_build/results?buildId=405915&view=logs&s=aee76c8a-1d8b-5149-1924-86de92a850c9

# Checklist (please delete before completing or setting auto-complete)
- [x] Story Work items associated (not Tasks)
- [x] Successful testing run(s) link provided
- [x] Title pattern should be `{work item number}: {title}`
- [x] Description covers all the changes in the PR
- [ ] This PR contains documentation
- [ ] This PR contains tests

# **How does this PR make you feel**:
gif

### CODE REVIEW

File: infra/core/key-vault/platform-key-vault.parameters.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
```
Comment (DaRoszko) [inline infra/core/key-vault/platform-key-vault.parameters.json:4]: as this file is **new**, the standard is bicepParameters for this one

**Review by DaRoszko — CHANGES_REQUESTED** [review]

File: infra/core/key-vault/platform-key-vault.parameters.json (hunk 1 above)
Comment (aadev1) [reply infra/core/key-vault/platform-key-vault.parameters.json:4]: Doh!  - I'll update it.

Comment (aadev1) [reply infra/core/key-vault/platform-key-vault.parameters.json]: @DaRoszko - I've converted it to bicep parameters and here's the successful build:

https://dev.azure.com/defragovuk/DEFRA-FFC/_build/results?buildId=406125&view=logs&j=6263ae07-080d-5a18-f654-fdb20c6ea0e3&t=6263ae07-080d-5a18-f654-fdb20c6ea0e3

**Review by ajaysoni-cap — DISMISSED** [review]

**Comment by aadev1** [comment]
Temporarily closing PR.  Will raise another one after making further changes

**Pull request closed without merging**